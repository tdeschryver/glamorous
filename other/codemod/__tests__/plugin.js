const fs = require('fs')
const path = require('path')
const babel = require('babel-core')

test('test everything in fixtures!', async () => {
  const files = await getFiles()

  expect.assertions(files.length)
  await Promise.all(
    files.map(async filename => {
      const stripped = path.basename(filename, path.extname(filename))
      const actual = await transformFileAsync(stripped)
      expect(actual).toMatchSnapshot(filename)
    }),
  )
})

function getFiles() {
  return new Promise((resolve, reject) => {
    fs.readdir(path.join(__dirname, './fixtures'), (err, result) => {
      if (err) {
        reject(err)
      }

      resolve(result)
    })
  })
}

function transformFileAsync(filename) {
  const options = {
    extends: path.join(__dirname, '../.babelrc'),
  }

  return new Promise((resolve, reject) => {
    babel.transformFile(
      path.join(__dirname, `./fixtures/${filename}.js`),
      options,
      (err, {code}) => {
        if (err) {
          reject(err)
        }

        resolve(code)
      },
    )
  })
}

// testing on this will always fail?
// function readFileAsync(filename) {
//   return new Promise((resolve, reject) => {
//     const file = path.join(__dirname,`./fixtures/${filename}-expected.js`)
//     fs.readFile(file, 'utf8', (err, result) => {
//       if (err) {
//         reject(err)
//       }

//       resolve(result)
//     })
//   })
// }
