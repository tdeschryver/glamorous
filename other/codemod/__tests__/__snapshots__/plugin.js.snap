// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`five.js 1`] = `
"// should transform, merge props (object) and theme (properties)
const Five = glamorous.div((
  {
    important,
    main
  }
) => ({
  fontSize: main.fontSize,
  color: important ? 'red' : 'black'
}))
"
`;

exports[`four.js 1`] = `
"// should transform, merge props (properties) and theme (object)
const Four = glamorous.div((
  {
    important,
    theme
  }
) => ({
  fontSize: theme.main.fontSize,
  color: important ? 'red' : 'black'
}))
"
`;

exports[`one.js 1`] = `
"// should transform, only use theme
const One = glamorous.div((
  {
    theme
  }
) => ({
  fontSize: theme.main.fontSize
}))
"
`;

exports[`seven.js 1`] = `
"// should not transform
const Seven = glamorous.div(({theme}) => ({
  fontSize: theme.main.fontSize,
}))
"
`;

exports[`six.js 1`] = `
"// should transform, merge props (properties) and theme (properties)
const Six = glamorous.div((
  {
    important,
    main
  }
) => ({
  fontSize: main.fontSize,
  color: important ? 'red' : 'black'
}))
"
`;

exports[`three.js 1`] = `
"// should transform, merge props (object) and theme (object) -> translate props object to keys
const Three = glamorous.div((
  {
    important,
    theme
  }
) => ({
  fontSize: theme.main.fontSize,
  color: important ? 'red' : 'black'
}))
"
`;

exports[`two.js 1`] = `
"
// should transform, only use theme and third argument stays as is
const Two = glamorous.div((
  {
    theme
  },
  tripple
) => ({
  fontSize: theme.main.fontSize
}))
"
`;
